public/main.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+let subReddits = [`https://www.reddit.com/r/Unity3D.json`, `https://www.reddit.com/r/blender.json`,
+`https://www.reddit.com/r/javascript.json`, `https://www.reddit.com/r/CivVI.json`,
+`https://www.reddit.com/r/Battletechgame.json`, `https://www.reddit.com/r/Cooking.json`, 
+`https://www.reddit.com/r/space.json`, `https://www.reddit.com/r/Guitar.json`, 
+`https://www.reddit.com/r/FullmetalAlchemist.json`, `https://www.reddit.com/r/Sculpture.json`,  ];
+let listingsArr = [];
+const postsElem =document.querySelector("#posts");
+const favButtonNames = ['#fav-button-one', '#fav-button-two', '#fav-button-three'];
+
+favButtonNames.forEach(function(name, index){
```

Comment:
Rather than using a for loop here. It would be easier to read if you just did three lines of: 

```
document.querySelector(name).addEventListener('click', () => requestListings(subReddits[0]));
document.querySelector(name).addEventListener('click', () => requestListings(subReddits[1]));
document.querySelector(name).addEventListener('click', () => requestListings(subReddits[2]));
```

To be even clearer, instead of using the subreddits array, you could just hard code the subreddit itself. 

Also, since `https://www.reddit.com/r/` is in each array item, you could pull that out in to a variable to be used in the requestListings function. Then the above code could look like: 

```
document.querySelector(name).addEventListener('click', () => requestListings('javascript'));
document.querySelector(name).addEventListener('click', () => requestListings('CivVI'));
document.querySelector(name).addEventListener('click', () => requestListings('Battletechgame'));
```

You could also just remove those three subreddits from the subreddits array. Then your random would not go to a subreddit that you already have a button for.